tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -240,32 +272,89 @@ static void xmit_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type
 
 	char *data = net_buf_simple_pull_mem(buf, sizeof(txt_msg));
 
+	LOG_INF("rx: %s", txt_msg);
 	ASSERT_EQUAL(0, memcmp(txt_msg, data, sizeof(txt_msg)));
 
-	if (cnt) {
-		int64_t delta = k_uptime_delta(&timestamp);
+	if (check_delta_time(xmit_param.retr, xmit_param.interval, 15)) {
+		LOG_INF("rx completed. stop observer.");
+		k_sem_give(&observer_sem);
+	}
+}
 
-		LOG_INF("rx: %s +%d ms", txt_msg, delta);
+static void rx_xmit_adv(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type       = BT_HCI_LE_SCAN_PASSIVE,
+		.options    = BT_LE_SCAN_OPT_NONE,
+		.interval   = BT_MESH_ADV_SCAN_UNIT(1000),
+		.window     = BT_MESH_ADV_SCAN_UNIT(1000)
+	};
+	int err;
 
-		ASSERT_TRUE(delta >= xmit_param.interval &&
-			delta < (xmit_param.interval + 15));
-	} else {
-		timestamp = k_uptime_get();
+	err = bt_le_scan_start(&scan_param, xmit_scan_cb);
+	ASSERT_FALSE(err && err != -EALREADY, "starting scan failed (err %d)", err);
+
+	err = k_sem_take(&observer_sem, K_SECONDS(20));
+	ASSERT_OK(err);
+
+	err = bt_le_scan_stop();
+	ASSERT_FALSE(err && err != -EALREADY, "stopping scan failed (err %d)", err);
+}
 
-		LOG_INF("rx: %s +0 ms", txt_msg);
+static void send_order_start_cb(uint16_t duration, int err, void *user_data)
+{
+	struct net_buf *buf = (struct net_buf *)user_data;
+
+	ASSERT_OK(err, "Failed adv start cb err (%d)", err);
+	ASSERT_EQUAL(2, buf->len);
+
+	uint8_t current = buf->data[0];
+	uint8_t previous = buf->data[1];
+
+	LOG_INF("tx start: current(%d) previous(%d)", current, previous);
+
+	ASSERT_EQUAL(previous_checker, previous);
+	previous_checker = current;
+}
+
+static void send_order_end_cb(int err, void *user_data)
+{
+	struct net_buf *buf = (struct net_buf *)user_data;
+
+	ASSERT_OK(err, "Failed adv start cb err (%d)", err);
+	ASSERT_TRUE(!buf->data, "Data not cleared!");
+	seq_checker++;
+	LOG_INF("tx end: seq(%d)", seq_checker);
+
+	if (seq_checker == num_adv_sent) {
+		seq_checker = 0;
+		previous_checker = 0xff;
+		k_sem_give(&observer_sem);
 	}
+}
 
-	cnt++;
+static void receive_order_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+			struct net_buf_simple *buf)
+{
+	uint8_t length;
+	uint8_t current;
+	uint8_t previous;
 
-	if (cnt > xmit_param.retr) {
-		cnt = 0;
-		timestamp = 0;
-		LOG_INF("rx completed. stop observer.");
+	length = net_buf_simple_pull_u8(buf);
+	ASSERT_EQUAL(buf->len, length);
+	ASSERT_EQUAL(BT_DATA_MESH_MESSAGE, net_buf_simple_pull_u8(buf));
+	current = net_buf_simple_pull_u8(buf);
+	previous = net_buf_simple_pull_u8(buf);
+	LOG_INF("rx: current(%d) previous(%d)", current, previous);
+	ASSERT_EQUAL(previous_checker, previous);
+
+	if (check_delta_time(xmit_param.retr, xmit_param.interval, xmit_param.interval)) {
```

Comment:
The third parameter for `check_delta_time` shall always be 15.

Reply:
If it is always 15 why is it required at all?